### DISCUSSION TITLE
Curiosity: How to build GraphQL APIs in ActiveJ?

### DISCUSSION THREAD

**Discussion by ghost** [activej/activej#161] [Q&A]
To mitigate under-fetching and over-fetching, we 're planning to test it.

**Comment by eduard-vasinskyi** [comment]
Hi, @kkr2606

You can use ActiveJ HTTP server in combination with Java’s GraphQL implementation. 

I have created a simple [hello-world demo](https://gist.github.com/eduard-vasinskyi/d8e8f4e2cecd4cb75932554be1365614) using ActiveJ with [graphql-java](https://github.com/graphql-java/graphql-java) library following guidelines from [GraphQL](https://graphql.org/learn/serving-over-http) and [graphql-java](https://www.graphql-java.com/documentation/getting-started/#hello-world) documentations.

A next logical step would be to abstract out GraphQL API into a separate [AsyncServlet](https://gist.github.com/eduard-vasinskyi/02d7fa40999b619a3d9bc54fdea4c17b)

[Here](https://gist.github.com/eduard-vasinskyi/7431e623662207b5f7e42da79ee46288) is how a hello-world demo would look like if used with a GraphQLServlet (with blocking call)

You can test demos by running curl commands:
```bash
curl -g http://localhost:8080/graphql?query={hello}
curl -X POST http://localhost:8080/graphql --data '{"query":"{hello}"}'
```

These demos can be used as a starting point for creating more complex GraphQL-based projects.

**Comment by ghost** [comment]
Thank you @eduard-vasinskyi. Let me try it out.
